unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -27,6 +27,105 @@ export function CategoryFacetTest() {
     let simulateQueryData: ISimulateQueryData;
 
+    describe('exposes options', () => {
+      describe('dependsOn', () => {
+        let master: IBasicComponentSetup<CategoryFacet>;
+        const masterFacetField = '@masterFacet';
+        const dependentFacetField = '@dependantFacet';
+        let simulateQueryDataMaster: ISimulateQueryData;
+        let simulateQueryDataDependant: ISimulateQueryData;
+
+        beforeEach(() => {
+          simulateQueryDataMaster = buildCategoryFacetResults(11, 11, masterFacetField);
+          simulateQueryDataDependant = buildCategoryFacetResults(11, 11, dependentFacetField);
+          master = Mock.advancedComponentSetup<CategoryFacet>(
+            CategoryFacet,
+            new Mock.AdvancedComponentSetupOptions(
+              undefined,
+              {
+                field: masterFacetField
+              },
+              env => env.withLiveQueryStateModel()
+            )
+          );
+
+          test = Mock.advancedComponentSetup<CategoryFacet>(
+            CategoryFacet,
+            new Mock.AdvancedComponentSetupOptions(
+              undefined,
+              {
+                field: dependentFacetField,
+                dependsOn: masterFacetField
+              },
+              env => env.withQueryStateModel(master.cmp.queryStateModel)
+            )
+          );
+        });
+
+        it('adds the coveo-facet-dependent class to the dependent facet', () => {
+          Simulate.query(test.env, simulateQueryDataDependant);
+          expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(true);
+        });
+
+        it('does not add the coveo-facet-dependent class to the master facet', () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          expect($$(master.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
+        });
+
+        it(`when the master facet has a selected value,
+        it removes the coveo-facet-dependent class from the dependent facet`, () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          master.cmp.selectValue('value9');
+          expect($$(test.cmp.element).hasClass('.coveo-facet-dependent')).toBe(false);
+        });
+
+        it(`when the master facet has a selected value,
+        it adds the coveo-category-facet-non-empty-path class to the master facet`, () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          master.cmp.selectValue('value9');
+          expect($$(master.cmp.element).hasClass('.coveo-category-facet-non-empty-path')).toBe(false);
+        });
+
+        describe('when the master and the depeandents have a selected value', () => {
+          function triggerStateChangeOnDependentFacet() {
+            rebindDependentFacetStateChangeListeners();
+            $$(test.env.root).trigger('change', { attributes: {} });
+          }
+
+          function rebindDependentFacetStateChangeListeners() {
```

Comment:
I have a feeling this is un-necessary...

Reply:
After some testing, I need `test.cmp['initQueryStateEvents']();`or else, the test failed as the value are not updated, I'll refactor a bit and check if I can call something else because normally, `initQueryStateEvents` should be called at the initialization